### PR TITLE
Fix Windows @scoped plugin resolution

### PR DESCRIPTION
**Pull request by oznu**

### CODE REVIEW

File: src/pluginManager.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export class PluginManager {
   }
 
   public static isQualifiedPluginIdentifier(identifier: string): boolean {
-    return PluginManager.PLUGIN_IDENTIFIER_PATTERN.test(identifier);
+    return PluginManager.PLUGIN_IDENTIFIER_PATTERN.test(identifier.replace(path.sep, "/"));
```
Comment (oznu) [inline src/pluginManager.ts:83]: Windows will use `\` in paths while the npm package names will always use `/` as the scope seperator. Package names cannot contain `\` so this won't conflict with anything else.

File: src/pluginManager.ts [hunk 2]
```diff
@@ -285,6 +285,7 @@ export class PluginManager {
 
             const absolutePath = path.join(searchPath, scopeDirectory);
             fs.readdirSync(absolutePath)
+              .filter(name => PluginManager.isQualifiedPluginIdentifier(name))
```
Comment (oznu) [inline src/pluginManager.ts:288]: Scoped packages can only be one level deep, this saves having to stat every directory in the @scope that cannot possibly be homebridge plugins.

File: src/pluginManager.ts (hunk 1 above)
Comment (Supereg) [reply src/pluginManager.ts:83]: I don't like the fact that we put that in to the isQualifiedPluginIdentifier function. And the backslash part is only really used in the context of plugin loading. Maybe we could add the replace just at the places this function is called in context of dealing with paths?

**Comment by Supereg** [comment]
Also the windows path style breaks the active plugin checks some lines below:
``` js
     && (!this.activePlugins || this.activePlugins.includes(pluginIdentifier)); // check if activePlugins is restricted and if so if the plugin is contained
```

File: src/pluginManager.ts [hunk 3]
```diff
@@ -285,8 +285,9 @@ export class PluginManager {
...
               .filter(name => fs.statSync(path.resolve(absolutePath, name)).isDirectory())
-              .forEach(name => relativePluginPaths.push(path.join(scopeDirectory, name)));
+              .forEach(name => relativePluginPaths.push(scopeDirectory + "/" + name));
```
Comment (Supereg) [inline src/pluginManager.ts:290]: By not using `path.join` and assembling the plugin by ourself and forcing a `/` we remove the need to have the replace thing in the PluginManager.isQualifiedPluginIdentifier function and also restore functionality for our activePlugins.includes check.

**Comment by Supereg** [comment]
Let me now what your opinion is on my proposed change. @oznu

**Pull request merged**